app/constants/device.ts

```diff
@@ -2,11 +2,18 @@
 // See LICENSE.txt for license information.
 
 import * as FileSystem from 'expo-file-system';
+import {Platform} from 'react-native';
 import DeviceInfo from 'react-native-device-info';
 
+const isPhoneWithInsets = Platform.OS === 'ios' && DeviceInfo.hasNotch();
+const isIPhone12Mini = DeviceInfo.getModel() === 'iPhone 12 mini';
+const isTablet = DeviceInfo.isTablet();
```

Comment:
Do we really need this?? it seems to me that we can do much better.

Also the isTablet should use the hook instead

Reply:
This is what is done on V1. I am all in for improving on this.

The usage of these constants is mainly to decide:
- How big is the Autocomplete box (bigger on tablet)
- How far from the top is the Autocomplete box (further on devices with a notch)

Showing the autocomplete when the keyboard is not visible may still be useful, since we may want "more space to select something from the autocomplete". Also it is useful for commands (you can start writing a command without opening the keyboard with the quick action).

We may have a logic to dismiss the autocomplete if we tap somewhere and the keyboard is hidden, but 0/5.

Reply:
We can keep the autocomplete with the keyboard closed, hopefully no bugs will be filed related to this.

As for the constants I would definitely want to improve on that:
1. isTablet (is defined by the hook)
2. instead of iPhoneWithInsets, mini, etc.. we can use the `useSafeAreaInsets` hook instead
3. For the size, the only place where I can see the autocomplete box being smaller is on a tablet in landscape mode with the keyboard opened as probably the available space from the post draft with multiple lines to the top of the screen is narrower